sagemaker-python-sdk/tensorflow_resnet_cifar10_with_tensorboard/utils.py

```diff
@@ -1,37 +1,25 @@
 import os
 import sys
 import tarfile
+import boto3
 from six.moves import urllib
 from ipywidgets import FloatProgress
 from IPython.display import display
 
-DATA_URL = 'https://www.cs.toronto.edu/~kriz/cifar-10-binary.tar.gz'
-
 
 def cifar10_download(data_dir='/tmp/cifar10_data', print_progress=True):
-    """Download and extract the tarball from Alex's website."""
     if not os.path.exists(data_dir):
         os.makedirs(data_dir)
 
     if os.path.exists(os.path.join(data_dir, 'cifar-10-batches-bin')):
         print('cifar dataset already downloaded')
         return
 
-    filename = DATA_URL.split('/')[-1]
+    filename = 'cifar-10-binary.tar.gz'
     filepath = os.path.join(data_dir, filename)
 
     if not os.path.exists(filepath):
-        f = FloatProgress(min=0, max=100)
-        display(f)
-        sys.stdout.write('\r>> Downloading %s ' % (filename))        
-
-        def _progress(count, block_size, total_size):
-            if print_progress:
-                f.value = 100.0 * count * block_size / total_size
-
-        filepath, _ = urllib.request.urlretrieve(DATA_URL, filepath, _progress)
-        print()
-        statinfo = os.stat(filepath)
-        print('Successfully downloaded', filename, statinfo.st_size, 'bytes.')
+        region = boto3.Session().region_name
+        boto3.Session().resource('s3').Bucket('sagemaker-sample-data-{}'.format(region)).download_file('tensorflow/cifar10/cifar-10-binary.tar.gz', '/tmp/cifar10_data/cifar-10-binary.tar.gz')
```

Comment:
this won't work in gov-cloud. They have access to regular http and https call, but not through s3 client because the regions are partitioned off.

Reply:
this isn't entirely accurate, but Dan's right that this won't work in GovCloud because the global S3 endpoint does not work in that region.

To make it work, specify the region when creating the boto resource:

```python
boto3.Session().resource('s3', region_name=region)
```

Reply:
Lauren is right.

Disregard my comment about the http and https calls.